### PR TITLE
Continue editing Chronograf OAuth docs

### PR DESCRIPTION
**Pull request by pierwill**
Work-in-progress.

Following up on previous work done in #2472.

### CODE REVIEW

File: content/chronograf/v1.7/administration/managing-security.md [hunk 1]
```diff
@@ -47,11 +49,11 @@ Set this value to a secure, arbitrary string.
 > ***InfluxEnterprise clusters:*** If you are running multiple Chronograf servers in a high availability configuration,
 > set the `TOKEN_SECRET` environment variable on each server to ensure that users can stay logged in.
 
-### JWKS Signature Verification
+### JSON Web Key Set (JWKS) Signature Verification (optional)
```
Comment (sanderson) [inline content/chronograf/v1.7/administration/managing-security.md]: Just use the acronym/initialism in headings. Just provide the unnabreviated form (with the abbreviation) in the body copy below the heading. Headings should be quickly scannable. They also get a unique html ID generated from the text of the heading that's used when linking directly to a heading/section.

File: content/chronograf/v1.7/administration/managing-security.md [hunk 2]
```diff
@@ -47,11 +49,11 @@ Set this value to a secure, arbitrary string.
...
 
-If the OAuth provider implements OpenID Connect with RS256 signatures (as Microsoft AD FS does),
-you need to enable `id_token` support and provide a JWKS document (holding the certificate chain), to validate the RSA signatures against.
-This certificate chain is regularly rolled over (when the certificates expire), so it is fetched from the JWKS_URL on demand.
+If the OAuth provider implements OpenID Connect with RS256 signatures, you need to enable this feature with the `USE_ID_TOKEN` variable
+and provide a JWKS document (holding the certificate chain), to validate the RSA signatures against.
```
Comment (sanderson) [inline content/chronograf/v1.7/administration/managing-security.md]: You shouldn't need the comma after the parentheses.

File: content/chronograf/v1.7/administration/managing-security.md [hunk 3]
```diff
@@ -47,11 +49,11 @@ Set this value to a secure, arbitrary string.
 > ***InfluxEnterprise clusters:*** If you are running multiple Chronograf servers in a high availability configuration,
 > set the `TOKEN_SECRET` environment variable on each server to ensure that users can stay logged in.
 
-### JWKS Signature Verification
+### JWKS Signature Verification (optional)
 
-If the OAuth provider implements OpenID Connect with RS256 signatures (as Microsoft AD FS does),
-you need to enable `id_token` support and provide a JWKS document (holding the certificate chain), to validate the RSA signatures against.
-This certificate chain is regularly rolled over (when the certificates expire), so it is fetched from the JWKS_URL on demand.
+If the OAuth provider implements OpenID Connect with RS256 signatures, you need to enable this feature with the `USE_ID_TOKEN` variable
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: must enable JWS signature verification? by setting the `USE...` variable to JSON...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 4]
```diff
@@ -72,17 +74,16 @@ Configuration steps for the following supported authentication providers are pro
 * [Auth0](#configure-auth0-authentication)
 * [Heroku](#configure-heroku-authentication)
 * [Okta](#configure-okta-authentication)
-* [Generic](#configure-generic-authentication)
+* [Gitlab](#configure-gitlab-authentication)
+* [Generic OAuth](#configure-generic-oauth-authentication)
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Generic OAuth provider?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 5]
```diff
@@ -72,17 +74,16 @@ Configuration steps for the following supported authentication providers are pro
...
 
 > If you haven't already, you must first [generate a token secret](#generate-a-token-secret) before proceeding.
 
-### Configure GitHub authentication
+---
 
-Chronograf supports using the [GitHub OAuth 2.0 authentication](https://developer.github.com/apps/building-oauth-apps/)
-to request authorization and provide authentication.
-To use GitHub authentication, you need to register a GitHub application and use the assigned Client ID and Client Secret.
+#### Configure GitHub authentication
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Will, should these "Configure..." headers  be bumped up to ### 3s?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 6]
```diff
@@ -117,7 +118,7 @@ export GH_CLIENT_SECRET=260041897d3252c146ece6b46ba39bc1e54416dc
 export GH_ORGS=biffs-gang
 ```
 
-#### Creating GitHub OAuth 2.0 applications
+##### Creating GitHub OAuth 2.0 applications
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Create

File: content/chronograf/v1.7/administration/managing-security.md [hunk 7]
```diff
@@ -117,7 +118,7 @@ export GH_CLIENT_SECRET=260041897d3252c146ece6b46ba39bc1e54416dc
...
 
 The following steps will guide you in configuring Chronograf to support GitHub OAuth 2.0 authorization and authentication.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Configure Chronograf to...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 8]
```diff
@@ -157,7 +158,7 @@ The following steps will guide you in configuring Chronograf to support GitHub O
     * [`--github-client-secret=`](/chronograf/v1.7/administration/config-options/#github-client-secret-s)
     * [`--token_secret=`](/chronograf/v1.7/administration/config-options/#token-secret-t)
 
-#### Optional GitHub organizations
+##### Optional GitHub organizations
 
 If you want to require a GitHub organization membership for a user, set the `GH_ORGS` environment variable.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: what will variable be set to?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 9]
```diff
@@ -196,7 +197,7 @@ Optionally, you can restrict access to specific domains using the following envi
 
 For details on Chronograf command line options and environment variables, see [Google OAuth 2.0 authentication options](/chronograf/v1.7/administration/config-options#google-specific-oauth-2-0-authentication-options).
 
-#### Creating Google OAuth 2.0 applications
+##### Creating Google OAuth 2.0 applications
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Create

File: content/chronograf/v1.7/administration/managing-security.md [hunk 10]
```diff
@@ -233,7 +234,7 @@ The following steps will guide you in configuring Google OAuth 2.0 authorization
     * [`--public-url=`](/chronograf/v1.7/administration/config-options/#public-url)
     * [`--token_secret=`](/chronograf/v1.7/administration/config-options/#token-secret-t)
 
-#### Optional Google domains
+##### Optional Google domains
 
 Similar to GitHub organization restrictions, Google authentication can be configured to restrict access to Chronograf to specific domains.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Configure Google auth...to restrict access to...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 11]
```diff
@@ -406,21 +407,20 @@ Your users should now be able to sign into Chronograf using the new Okta provide
...
-    --generic-api-url=http://gitlab-server-example.com/api/v3/user
-    --public-url=http://chronograf-example.com:8888/
+    --generic-api-url=https://gitlab.com/api/v3/user
+    --public-url=http://<chronograf-host>:8888/
     ```
 
 3. Restart the Chronograf service.
 
 Your users should now be able to sign into Chronograf using the new GitLab provider.
+#### Configure generic OAuth authentication
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: maybe drop line 421/one heading is enough? Configure Chronograf to use any OAuth 2.0 provider

File: content/chronograf/v1.7/administration/managing-security.md (hunk 5 above)
Comment (pierwill) [reply content/chronograf/v1.7/administration/managing-security.md]: I was thinking they fall _under_ the ###-level heading "OAuth 2.0 providers". Do you think they'd look better as higher-level headings on par with the provider list?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 12]
```diff
@@ -9,11 +9,12 @@ menu:
...
 ---
 
-To enhance security, configure Chronograf to authenticate and authorize with [OAuth 2.0](https://oauth.net/) and TLS/HTTPS.
+To enhance security, configure Chronograf to authenticate and authorize with [OAuth 2.0](https://oauth.net/) and use TLS/HTTPS.
 
 * [Configure OAuth 2.0](#configure-oauth-2-0)
   1. [Generate a Token Secret](#generate-a-token-secret)
   2. [Set configurations for your OAuth provider](#oauth-2-0-providers)
+  3. [Configure authentication duration](#configure-authentication-duration)
 * [Configure TLS (Transport Layer Security) and HTTPS](#configure-tls-transport-layer-security-and-https)
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: given "use" in line 12...should line 17 be "Use

Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: w "use" in line 12...should line 18 be "Use TLS....

File: content/chronograf/v1.7/administration/managing-security.md [hunk 13]
```diff
@@ -22,8 +23,9 @@ To enhance security, configure Chronograf to authenticate and authorize with [OA
 > You can then set up [multiple organizations](https://docs.influxdata.com/chronograf/latest/administration/managing-organizations/)
 > and [users](https://docs.influxdata.com/chronograf/latest/administration/managing-influxdb-users/).
 
-Chronograf can use [OAuth 2.0](https://tools.ietf.org/html/rfc6749) providers and [JWT (JSON Web Token)](https://tools.ietf.org/html/rfc7519) tokens
-to require authorization and authentication of Chronograf users and enable role-based access controls.
+Configure Chronograf to use an OAuth 2.0 provider and JWT (JSON Web Token) token to authenticate users and and enable role-based access controls.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: maybe drop 2nd token?

Comment (kelseiv) [reply content/chronograf/v1.7/administration/managing-security.md]: double and

File: content/chronograf/v1.7/administration/managing-security.md [hunk 14]
```diff
@@ -72,65 +74,25 @@ Configuration steps for the following supported authentication providers are pro
...
+---
 
-The following steps will guide you in configuring Chronograf to support GitHub OAuth 2.0 authorization and authentication.
+#### Configure GitHub authentication
 
-**To create a GitHub OAuth 2.0 application:**
+First create a GitHub OAuth 2.0 application:
 
 1. Follow the [Register your app](https://github.com/settings/applications/new) steps at the GitHub Developer website to register your GitHub application and obtain your assigned Client ID and Client Secret.
+   On the GitHub application registration page, you will need to enter the following values:
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: page, enter

File: content/chronograf/v1.7/administration/managing-security.md [hunk 15]
```diff
@@ -72,65 +74,25 @@ Configuration steps for the following supported authentication providers are pro
...
-```
-
-#### Creating GitHub OAuth 2.0 applications
+---
 
-The following steps will guide you in configuring Chronograf to support GitHub OAuth 2.0 authorization and authentication.
+#### Configure GitHub authentication
 
-**To create a GitHub OAuth 2.0 application:**
+First create a GitHub OAuth 2.0 application:
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Is there a second part?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 16]
```diff
@@ -72,65 +74,25 @@ Configuration steps for the following supported authentication providers are pro
...
 
 1. Follow the [Register your app](https://github.com/settings/applications/new) steps at the GitHub Developer website to register your GitHub application and obtain your assigned Client ID and Client Secret.
+   On the GitHub application registration page, you will need to enter the following values:
+   - **Homepage URL**: the full Chronograf server name and port.
+   If you are running it locally with the default settings, it would be `http://localhost:8888`.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: For example, to run the app locally w default settings, set the URL to `http:....

File: content/chronograf/v1.7/administration/managing-security.md [hunk 17]
```diff
@@ -72,65 +74,25 @@ Configuration steps for the following supported authentication providers are pro
...
+   - **Authorization callback URL**: the **Homepage URL** plus the callback URL path `/oauth/github/callback`
+   (e.g. `http://localhost:8888/oauth/github/callback`).
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: for example, `h...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 18]
```diff
@@ -151,56 +113,65 @@ The following steps will guide you in configuring Chronograf to support GitHub O
     export TOKEN_SECRET=Super5uperUdn3verGu355!
     ```
 
-    Alternatively, these environment variables can be set using the equivalent command line options:
+Alternatively, these environment variables can be set using the equivalent command line options:
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Alternatively, set env vars using the...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 19]
```diff
@@ -151,56 +113,65 @@ The following steps will guide you in configuring Chronograf to support GitHub O
...
+* [`--github-client-secret=`](/chronograf/v1.7/administration/config-options/#github-client-secret-s)
+* [`--token_secret=`](/chronograf/v1.7/administration/config-options/#token-secret-t)
+
+For details on the command line options and environment variables, see [GitHub OAuth 2.0 authentication options](/chronograf/v1.7/administration/config-options#github-specific-oauth-2-0-authentication-options).
 
-#### Optional GitHub organizations
+##### GitHub organizations (optional)
 
-If you want to require a GitHub organization membership for a user, set the `GH_ORGS` environment variable.
+If you want to require a GitHub organization membership for a user, set the `GH_ORGS` environment variable with the name of your organization.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: To require

File: content/chronograf/v1.7/administration/managing-security.md [hunk 20]
```diff
@@ -151,56 +113,65 @@ The following steps will guide you in configuring Chronograf to support GitHub O
...
 
-#### Overview
+##### Overview
 
-Chronograf supports the use of the [Google OAuth 2.0 protocol](https://developers.google.com/identity/protocols/OAuth2) for authentication and authorization.
 The steps below guide you in creating the following required Chronograf environment variables (or corresponding command line options):
 
 * `GOOGLE_CLIENT_ID` (Google Client ID)
 * `GOOGLE_CLIENT_SECRET` (Google Client Secret)
 * `PUBLIC_URL` (Public URL)
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: To create the following required Chronograf...., create a Google OAuth....

File: content/chronograf/v1.7/administration/managing-security.md [hunk 21]
```diff
@@ -151,56 +113,65 @@ The following steps will guide you in configuring Chronograf to support GitHub O
...
-
 For details on Chronograf command line options and environment variables, see [Google OAuth 2.0 authentication options](/chronograf/v1.7/administration/config-options#google-specific-oauth-2-0-authentication-options).
 
-#### Creating Google OAuth 2.0 applications
-
-The following steps will guide you in configuring Google OAuth 2.0 authorization and authentication support in Chronograf.
+##### Create a Google OAuth 2.0 application
 
-**To create a Google OAuth 2.0 application:**
+To create a Google OAuth 2.0 application:
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: I'd prob drop line 174

File: content/chronograf/v1.7/administration/managing-security.md [hunk 22]
```diff
@@ -233,22 +204,23 @@ The following steps will guide you in configuring Google OAuth 2.0 authorization
...
 ```
 
-### Configure Auth0 authentication
+#### Configure Auth0 authentication
 
 [Auth0](https://auth0.com/) implements identity protocol support for OAuth 2.0.
 See [OAuth 2.0](https://auth0.com/docs/protocols/oauth2) for details about the Auth0 implementation.
 
-#### Creating an Auth0 application
+##### Creating an Auth0 application
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: maybe use the same conjugation throughout the headers?

File: content/chronograf/v1.7/administration/managing-security.md [hunk 23]
```diff
@@ -233,22 +204,23 @@ The following steps will guide you in configuring Google OAuth 2.0 authorization
...
 
 The following steps will guide you in configuring Auth0 OAuth 2.0 authorization and authentication support in Chronograf.
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: I'd prob search and replace "The following steps will guide you in configuring" >> "Configure...

File: content/chronograf/v1.7/administration/managing-security.md [hunk 24]
```diff
@@ -406,33 +378,31 @@ Your users should now be able to sign into Chronograf using the new Okta provide
...
+When using the generic configuration, some or all of the following environment variables (or corresponding command line options) are required (depending on your OAuth 2.0 provider):
 
 * `GENERIC_CLIENT_ID`: Application client [identifier](https://tools.ietf.org/html/rfc6749#section-2.2) issued by the provider
 * `GENERIC_CLIENT_SECRET`: Application client [secret](https://tools.ietf.org/html/rfc6749#section-2.3.1) issued by the provider
 * `GENERIC_AUTH_URL`: Provider's authorization [endpoint](https://tools.ietf.org/html/rfc6749#section-3.1) URL
 * `GENERIC_TOKEN_URL`: Provider's token [endpoint](https://tools.ietf.org/html/rfc6749#section-3.2) URL used by the Chronograf client to obtain an access token
 * `USE_ID_TOKEN`: Enable OpenID [id_token](https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.3.1.3.3) processing
-* `JWKS_URL`: OAuth 2.0 provider's jwks [endpoint](https://tools.ietf.org/html/rfc7517#section-4.7) is used by the client to validate RSA signatures
+* `JWKS_URL`: OAuth 2.0 provider's JWKS [endpoint](https://tools.ietf.org/html/rfc7517#section-4.7) is used by the client to validate RSA signatures
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: ..) used by

File: content/chronograf/v1.7/administration/managing-security.md [hunk 25]
```diff
@@ -22,8 +23,9 @@ To enhance security, configure Chronograf to authenticate and authorize with [OA
 > You can then set up [multiple organizations](https://docs.influxdata.com/chronograf/latest/administration/managing-organizations/)
```
Comment (kelseiv) [inline content/chronograf/v1.7/administration/managing-security.md]: Then, you can...

**Pull request merged**